### PR TITLE
feat(collections): add maxWith

### PR DESCRIPTION
**Pull request by majidsajadi**
this PR adds `maxWith` to the `collections` module.

### CODE REVIEW

File: collections/max_with.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns the first element having the largest value according to the provided comparator or undefined if there are no elements
```
[the remark this answers is not in the document]
Comment (LionC) [reply collections/max_with.ts:4]: @majidsajadi I think if you update this we can merge it :-)

[the remark this answers is not in the document]
Comment (majidsajadi) [reply collections/max_with.ts]: applied your suggestion. let me know if I should change anything else. @LionC

[the remark this answers is not in the document]
Comment (LionC) [reply collections/max_with.ts]: @majidsajadi I think you forgot to update the README as well

**Review by kt3k — APPROVED** [review]
@majidsajadi LGTM. Thank you for your contribution!

@LionC Thank you for your reviews!

**Pull request merged**